File: Program75.java
import java.lang.*;
import java.util.*;

class ArrayX
{
    private int Arr[];
    private int iNo;

    public ArrayX(int iValue)
    {
        Arr = new int[iValue];
    }
    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println("Enter element: "+(iCnt+1));
            Arr[iCnt] = sobj.nextInt();
        }

        System.out.println("Enter element to search");
        iNo = sobj.nextInt();
    }
    public int SearchLastOccurance()
    {
        int iCnt = 0;
        
        for(iCnt = (Arr.length-1); iCnt >= 0; iCnt--)
        {
            if(Arr[iCnt] == iNo)
            { 
                break;
            }
        }
        return iCnt;
    }
}

class Program75
{
    public static void main(String arg[])
    {
       Scanner sobj = new Scanner(System.in);
       int iLength = 0; int iRet = 0;

       System.out.println("Enter the value of elements");
       iLength = sobj.nextInt();

       ArrayX aobj = new ArrayX(iLength);

       aobj.Accept();

       iRet = aobj.SearchLastOccurance();
       
       if(iRet == -1)
       {
           System.out.println("There is no such element in Array");
       }
       else
       {
        System.out.println("Element Last occured at: "+iRet);
       }

    }
}

File: Program75.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

int SearchLastOccurance(int Arr[], int iLength, int iNo)
{
    int iCnt = 0;

    for(iCnt = iLength-1; iCnt >= 0; iCnt--)
    {
        if(Arr[iCnt] == iNo)
        {
            break;
        }
    }
    return iCnt;
}

int main()
{
    int iCnt = 0, iSize = 0, iValue = 0;
    int *ptr = NULL;
    int iRet = 0;

    printf("Enter number of elements\n");
    scanf("%d",&iSize);

    ptr = (int*)malloc(iSize * sizeof(int));

    if(ptr == NULL)
    {
        printf("Unable to allocate memory");
        return -1;
    }

    printf("Entered element is: %d\n",iSize);

    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter element: %d\n",iCnt+1);
        scanf("%d",&ptr[iCnt]);
    }

    printf("Enter element to search\n");
    scanf("%d",&iValue);

    iRet = SearchLastOccurance(ptr, iSize, iValue);

    if(iRet == -1)
    {
        printf("There is no such element");
    }
    else
    {
        printf("Element Last occured at: %d",iRet);
    }
}